embulk-standards/src/main/java/org/embulk/standards/JsonParserPlugin.java

```diff
@@ -99,12 +131,79 @@ private PageBuilder newPageBuilder(Schema schema, PageOutput output)
         return new PageBuilder(Exec.getBufferAllocator(), schema, output);
     }
 
-    private JsonParser.Stream newJsonStream(FileInputInputStream in)
-            throws IOException
-    {
-        return new JsonParser().open(in);
+    private JsonParser.Stream newJsonStream(FileInputInputStream in , PluginTask task)
+            throws IOException {
+        InvalidEscapeStringPolicy policy = task.getInvalidEscapeStringPolicy();
+        switch (policy) {
+            case SKIP:
+            case UNESCAPE:
+                Iterable<CharSource> lines = Lists.transform(CharStreams.readLines(new BufferedReader(new InputStreamReader(in))),
+                        invalidEscapeStringFunction(policy));
+                return new JsonParser().open(new ReaderInputStream(CharSource.concat(lines).openStream()));
+            case PASSTHROUGH:
+            default:
+                return new JsonParser().open(in);
+        }
+    }
+
+    Function<String, CharSource> invalidEscapeStringFunction(final InvalidEscapeStringPolicy policy) {
+        return new Function<String, CharSource>() {
+            final Pattern p = Pattern.compile("\\p{XDigit}+");
+            @Override
+            public CharSource apply(@Nullable String input) {
+                assert input != null;
+                if (policy == InvalidEscapeStringPolicy.PASSTHROUGH) {
+                    return CharSource.wrap(input);
+                }
+                StringBuilder s = new StringBuilder();
+                char[] charArray = input.toCharArray();
+                for (int i=0 ; i < charArray.length ; i++) {
+                    char c = charArray[i];
+                    if (c == '\\') {
+                        if (charArray.length > i + 1) {
+                            char next = charArray[i + 1];
+                            switch (next) {
+                                case 'b':
+                                case 'f':
+                                case 'n':
+                                case 'r':
+                                case 't':
+                                case '"':
+                                case '\\':
+                                case '/':
+                                    s.append(c);
+                                    break;
+                                case 'u': // hexstring such as \u0001
+                                    if (charArray.length > i + 5) {
```

Comment:
Sorry, my thought did not reach 1 or 4 and it's better than 2.
In my opinion of SKIP mode would be **1**. 

Because In 4 mode, if 12xY of "12" is a valid part of a sentence, it will breaking removing "12". 
How do you think this idea?